grin/src/AbstractInterpretation/LiveVariable/CodeGenBase.hs

```diff
@@ -172,8 +184,11 @@ codeGenType cgSimpleTy cgNodeTy = \case
 isPointer :: IR.Predicate
 isPointer = IR.ValueIn (IR.Range 0 (maxBound :: Int32))
 
-isNotPointer :: IR.Predicate
-isNotPointer = IR.ValueIn (IR.Range (minBound :: Int32) 0)
+isLivenessInfo :: IR.Predicate
+isLivenessInfo = IR.ValueIn (IR.Range (-1) 0)
```

Comment:
I suggest to code  `_SOME_NAME_ = (-1)`  to these constants. It improves the code readability and maintainability.

Reply:
In the `CodeGen` module, where they are used heavily, I did give them a name. Here, I didn't feel necessary, since the name of the function is quite descriptive. In my opinion, it would be more confusing to name these constants here, since we are talking about ranges. Something like `livenessLowerBound` would make much sense since that info is already readable from the definition. I'm open to other suggestions.

Reply:
Ok, this is an implementation detail. But what is the interpretation of the (-1) in this particular case?

Reply:
`(-1)` is the representation of abstract value "live". `(-2)` is the same for "has effect".
I will think of a name. Do you have any suggestions?

Reply:
It is dead simple: _LIVE_ and _HAS_EFFECT_ .